src/background/advisory/index.js

```diff
@@ -23,13 +24,24 @@ export default async (packageID) => {
       stars,
     };
   });
+  const results = await Promise.allSettled([
+    fetchDebricked(normalizedPackageID),
+    fetchSnyk(normalizedPackageID),
+    fetchSocket(normalizedPackageID),
+    fetchOpenbase(normalizedPackageID),
+  ]);
+  const fulfilledResultsNormalized = results.reduce((acc, currentRes, index) => {
+    if (currentRes.status === 'fulfilled') {
+      acc[thirdPartyPackages[index]] = Promise.resolve(currentRes.value);
+    } else {
+      console.log({ failedPackage: [thirdPartyPackages[index]], reason: currentRes.reason });
+    }
+    return acc;
+  }, {});
```

Comment:
I understand your change, all this code was written quickly and dirty, so feel free to re-write it from scratch, there is no "best practices" here you missed ;-).

Still, few comments:

1. I don't like the weak relation between the `thirdPartyPackages` array and the `results` array, both based on their index. It's mean that you always have to be aware to change the other when you change one.
  My suggestion is to create an object, a map from advisroy name to its fetch function, so that you can build a consistent array from it.
1. Note that you changed the return value of the function a little. Before your change, if there was an error, the advisory was returned with `null`, and now with your change it will not be returned at all. Just need to make sure it will not break something.
1. Please add the `depsDev` to the `Promise.allSettled`.

Reply:
Also the handleAsyncError wouldnt work with fetch as when fetch fails it doesn't throw error when the error status is 4XX or 5XX. you have to catch it with res.ok.
https://stackoverflow.com/questions/38235715/fetch-reject-promise-and-catch-the-error-if-status-is-not-ok

Reply:
1. I will change it to be more readable and coupled.
2. shouldn't be a problem as in background-events.js listener function gets the object of promises and extract the entries. so if the is nothing there it will not extract it.